src/test/java/dev/openfga/sdk/errors/FgaErrorIntegrationTest.java

```diff
@@ -0,0 +1,340 @@
+package dev.openfga.sdk.errors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import dev.openfga.sdk.api.configuration.ClientConfiguration;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpHeaders;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Integration tests that verify the complete error handling flow,
+ * especially the getMessage() override behavior when wrapped in ExecutionException.
+ */
+class FgaErrorIntegrationTest {
+
+    @Test
+    void testErrorMessageVisibilityInExecutionException() throws Exception {
+        // Given - Simulate a validation error from the API
```

Comment:
We could test these against an OpenFGA server by moving them to be part of the [integration tests](https://github.com/openfga/java-sdk/blob/main/src/test-integration/java/dev/openfga/sdk/api/client/OpenFgaClientIntegrationTest.java). That way, we wouldn't need to maintain the error messages here in the event that they might change?

Reply:
You're right! Let me do that